pkg/command/command.go

```diff
@@ -71,29 +84,101 @@ func Elevate(password string) {
 
 	// Open STDIN
 	stdin, err := cmd.StdinPipe()
-	errorcheck.ErrorCheck(err, "\nFailed to get sudo STDIN")
+	common.ErrorCheck(err, "\nFailed to get sudo STDIN")
 
 	// Start the authentication
-	cmd.Start()
+	err = cmd.Start()
+	common.ErrorCheck(err, "\nFailed to start sudo command")
 
 	// Get the passed password
 	pw, _ := base64.StdEncoding.DecodeString(password)
 	_, err = stdin.Write([]byte(string(pw) + "\n"))
-	errorcheck.ErrorCheck(err, "\nFailed at typing to STDIN")
+	common.ErrorCheck(err, "\nFailed at typing to STDIN")
 	// Clear the password
 	pw = nil
 	password = ""
 
-	stdin.Close()
+	_ = stdin.Close()
 
 	// Wait for the sudo prompt (If the correct password was given, it will not stay behind)
 	err = cmd.Wait()
-	errorcheck.ErrorCheck(err, "\nError, password given was wrong")
+	common.ErrorCheck(err, "\nError, password given was wrong")
 }
 
-// Function to just clear the terminal
+// Clear clears the terminal.
 func Clear() {
 	c := exec.Command("clear")
 	c.Stdout = os.Stdout
-	c.Run()
+	_ = c.Run()
+}
+
+func processCmdString(cmd string) (string, []string) {
+	// handle quoted arguments
+	args := strings.Fields(cmd)
+	cmdBin := args[0]
+	args = args[1:]
+	for i, arg := range args {
+		if !strings.HasPrefix(arg, "\"") {
+			continue
+		}
+		// find the end of the quoted argument
+		for j, a := range args[i:] {
+			if strings.HasSuffix(a, "\"") {
+				args[i] = strings.Join(args[i:i+j+1], " ")
+				args = append(args[:i+1], args[i+j+1:]...)
+				break
```

Comment:
After seeing this commit, i remembered i already fixed this in `command.Run`
a less complex way would be to just pass any parameters after cmd as args that get combined into a stringlist
```go
func ExecAndLogSudo(isRoot, noisy bool, binary string, args ...string) error {
```

this means we could use
```go
ExecAndLogSudo(isRoot, true, "kernelstub", "-a", kernel_args)
```
`kernelstub` will be the cmd, `-a` and the `kernel_args` string will get combined into a `[]string` that you can just expand the args when we call exec.Command `exec.Command(binary, args...)`, no processing needed.
See the old implemention in `func Run` in command.go :)

that is how the old implementation did it and it worked and didnt rely on processing the args and using quotes since every parameter after the cmd was an argument.

We could instead just make an if to see if we are root and then "promote" the `binary` variable to  an arg
`exec.Command("sudo", binary, args...)`
and if we are root just do
`exec.Command(binary, args...)`

i believe this would be much cleaner and i feel dumb for not even thinking about it on my last comment 🤦

Reply:
actually you just did this with `RunErrSudo` so just need to do the same thing for `ExecAndLogSudo`

Reply:
I don't disagree that there are better and simpler ways to do this, and this isn't the only area that statement rings true for. Out of curiosity, is it not working reliably as it sits?

Reply:
> I don't disagree that there are better and simpler ways to do this, and this isn't the only area that statement rings true for. Out of curiosity, is it not working reliably as it sits?
> 
> In situations where a solution has unnecessary complexity I do my best to compensate with test coverage. I am not against refactoring it further but if I'm being honest if we were to get particular about simpler and more correct solutions we could be here all year 😅
> 
> That said, if it is currently still not working then I'll definitely jump back on it. (I will either way if you insist, of course)

Personally i think the simpler method just using stringlists would be the best and i am sorry you had to spend time on that other workaround.

I also feel it will be a lot safer than forcing in quotes and stripping them  (potentially creating unintended behavior) since every entry in a `[]string` is treated as if it is quoted if it contains spaces in terms of `exec.Command`.

I can merge as is and manually fix it myself as what you have already done is herculean for me and i really appreciate it! But i may not be able to write a test for it without me sitting down for a day and reading up properly on it (again).
Then i will just tag a new release once it is done. :smiley:

Reply:
I certainly can't say I disagree with your points here. You don't need to be sorry, it was just the closest roll of proverbial duct tape I conjured up. I'll take a look into this

Reply:
@HikariKnight 
latest changes should do it! added a much more realistic unit test to verify the behavior as well

Reply:
thanks i will run a check later today if i have the time, i just got back from vacation :smile:

Reply:
Awesome, no worries!